test/puppetlabs/puppetdb/testutils.clj

```diff
@@ -219,54 +219,74 @@
               :body body
               :params params))))
 
+(defn query-request
+  "Create a ring request map for a PuppetDB query. `http-method` indicates :get or :post.
+  Parameters or headers are optional"
```

Comment:
I don't seem to be able to curl successfully without headers. Is that what you meant here?